Src/VimCore/MarkMap.fs

```diff
@@ -92,6 +94,27 @@ type MarkMap(_bufferTrackingService : IBufferTrackingService) =
             vimBufferData.VimTextBuffer.GetLocalMark localMark
         | Mark.LastJump -> 
             vimBufferData.JumpList.LastJumpLocation
+        | Mark.LastExitedPosition ->
+            let getZeroZero() =
+                let snapshotLine' = vimBufferData.TextBuffer.CurrentSnapshot.GetLineFromPosition 0
+                let snapshot = snapshotLine'.Start.Add(0)
+                Some (VirtualSnapshotPointUtil.OfPoint snapshot)
+
+            let (line, column) =
+                match _globalLastExitedMap.TryFind (vimBufferData.Vim.VimHost.GetName vimBufferData.TextBuffer) with
+                | Some (line, column) -> (line, column)
+                | None -> (0, 0)
+
+            // not using TryGetPointInLine because None is returned when column and lineLength are both 0,
+            //  which can be valid in this case
+            match SnapshotUtil.TryGetLine vimBufferData.TextBuffer.CurrentSnapshot line with
+            | None -> getZeroZero()
+            | Some snapshotLine ->
+                if column >= snapshotLine.Length && not (column = 0 && snapshotLine.Length = 0) then
+                    getZeroZero()

```

Comment:
Consider the case where `line` is a non-zero value and `column` is the end position on the line.  The function will then return the start position of the entire buffer instead of the current line.  Is that behavior intentional?